Src/Support/Google.Apis.Auth.AspNetCore3.IntegrationTests/ClientInfo.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+Copyright 2018 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Google.Apis.Auth.AspNetCore.IntegrationTests
+{
+    /// <summary>
+    /// Client auth information, loaded from a Google user credential json file.
+    /// Set the TEST_CLIENT_SECRET_FILENAME environment variable to point to the credential file.
+    /// </summary>
+    public class ClientInfo
+    {
+        public static ClientInfo Load()
+        {
+            const string ClientSecretFilenameVariable = "TEST_CLIENT_SECRET_FILENAME";
+            string clientSecretFilename = Environment.GetEnvironmentVariable(ClientSecretFilenameVariable);
+            if (string.IsNullOrEmpty(clientSecretFilename))
+            {
+                throw new InvalidOperationException($"Please set the {ClientSecretFilenameVariable} environment variable before running tests.");
+            }
+            // This MUST be a "web" credential, not an "installed app" credential.
+            var secrets = JObject.Parse(Encoding.UTF8.GetString(File.ReadAllBytes(clientSecretFilename)))["web"];
```

Comment:
Use `File.ReadAllText` instead of reading all the bytes and then converting?

Reply:
Done